### PR TITLE
add tests for operator log

### PR DESCRIPTION
**Pull request by xiancao**
add the tests for verifying operator logs contains useful information and server msgs when there is a failure.
Refer to https://jira.oraclecorp.com/jira/browse/OWLS-76681

Jenkins result:
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/5370/
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/5372/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5396/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5395/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 1]
```diff
@@ -998,14 +1001,29 @@ public void testOnNonDynamicChangesCommitUpdateOnly() {
 
   }
 
+  @Test
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:1004]: describe the usecase with the reason for for introspection failure.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 2]
```diff
@@ -275,6 +278,18 @@ public static void initAll(@Namespaces(5) List<String> namespaces) {
     }
   }
 
+  @Test
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:281]: describe the usecase in JAVADOC

**Review by anpanigr — COMMENTED** [review]
Can you add few Negative tests that generate SEVERE message on Operator,  you can  add some negative usecases, where the domain does not comes up and the operator put proper SEVERE message on Log for better triaging the issue

**Review by sankarpn — APPROVED** [review]
LGTM

**Review by anpanigr — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 3]
```diff
@@ -1522,4 +1546,101 @@ private void waitForFileExistsInPod(String namespace, String podName, String fil
...
+
+  /**
+   * Negative test case for creating a model-in-image domain without encryption secret created.
+   * The admin server service/pod will not be created.
+   * The error message should be logged in the operator log.
+   *
+   * @param domainUid the uid of the domain to be created
+   * @param domainNamespace namespace in which the domain will be created
+   */
+  private void createMiiDomainNegative(String domainUid, String domainNamespace) {
```
Comment (rjeberhard) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1558]: This test seems unrelated to operator logging.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 4]
```diff
@@ -1522,4 +1546,101 @@ private void waitForFileExistsInPod(String namespace, String podName, String fil
...
+
+    setPodAntiAffinity(domain);
+
+    // create model in image domain
+    logger.info("Creating model in image domain {0} in namespace {1} using docker image {2}",
+        domainUid, domainNamespace, miiImage);
+    createDomainAndVerify(domain, domainNamespace);
+
+    // sleep for a while and check that the admin server service does not exist since the encryption secret not created
+    try {
```
Comment (rjeberhard) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1618]: I don't think that we should be relying on waiting for a set time period. This makes tests slower than we need to be. The typical pattern is to wait for a maximum of some amount of time while waiting for a specific condition. In this case, it is likely a failure status on the domain.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1558]: This is to create a negative test case to generate SEVERE log msg in the operator log.

Comment (rjeberhard) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1558]: Okay, I don't see that assertion. Is it already in checkServiceDoesNotExist()?

Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1558]: No, checkServiceDoesNotExist() mainly to wait and check the service is deleted and not existing any more.

Comment (rjeberhard) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1558]: Where is the check of the SEVERE in the log?

Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1558]: It is in the test method testOperatorLogSevereMsg().

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java]: fixed
